api/src/main/java/org/openmrs/module/serialization/xstream/XStreamSerializer.java

```diff
@@ -96,81 +110,78 @@ public XStreamSerializer() throws SerializationException {
+		this.customXStream = customXStream;
+	}
+
+	/**
+	 * Called by getXstream() during lazy-initialization of XStream.
+	 * 
+	 * @param xstream
+	 */
+	protected void initXStream(XStream xstream) {
 		// config the basic attributes
-		
+
 		// BaseOpenmrsObject
 		xstream.useAttributeFor(BaseOpenmrsObject.class, "uuid");
-		
+
 		// BaseOpenmrsData
 		xstream.useAttributeFor(BaseOpenmrsData.class, "voided");
-		
+
 		// BaseOpenmrsMetadata
 		xstream.useAttributeFor(BaseOpenmrsMetadata.class, "retired");
-		
+
 		// Other classes has voided or retired property
 		xstream.useAttributeFor(Concept.class, "retired");
 		xstream.useAttributeFor(ConceptName.class, "voided");
 		xstream.useAttributeFor(ConceptNameTag.class, "voided");
 		//xstream.useAttributeFor(ConceptSource.class, "retired");
-		
+
 		/*
 		 * alias className for all classses current need to serialize
 		 */
-		this.commonConfig();
-		
+		try {
+			aliasClassNames(xstream);
+		} catch (SerializationException e) {
+			throw new RuntimeException(e);
+		}
+
 		// CustomReflectionConverter to avoid the exception thrown when xstream deserialize a unknown elment
 		xstream.registerConverter(new CustomReflectionConverter(xstream.getMapper(), xstream.getReflectionProvider()),
-		    xstream.PRIORITY_LOW);
-		
+			xstream.PRIORITY_LOW);
+
 		// register this converter, so that we can let xstream serialize User.user only as its uuid
 		xstream.registerConverter(new UserConverter(xstream));
 		xstream.registerConverter(new CustomCGLIBEnhancedConverter(xstream.getMapper(), xstream.getConverterLookup()));
 		xstream.registerConverter(new CustomJavassistEnhancedConverter(xstream.getMapper(), xstream.getConverterLookup()));
 		xstream.registerConverter(new CustomSQLTimestampConverter());
 		xstream.registerConverter(new DateConverter("yyyy-MM-dd HH:mm:ss z", new String[] { "yyyy-MM-dd HH:mm:ss.S z",
-		        "yyyy-MM-dd HH:mm:ssz", "yyyy-MM-dd HH:mm:ss.S a", "yyyy-MM-dd HH:mm:ssa" }));
+			"yyyy-MM-dd HH:mm:ssz", "yyyy-MM-dd HH:mm:ss.S a", "yyyy-MM-dd HH:mm:ssa" }));
 
 		xstream.registerConverter(new CustomDynamicProxyConverter(), XStream.PRIORITY_VERY_HIGH);
 		// set our own defined marshalling strategy so that we can build references for cglib
 		xstream.setMarshallingStrategy(new CustomReferenceByIdMarshallingStrategy());
-	}
 
-    @PostConstruct
-    private void init(){
 		/*
 		 * Converters so that we can better deal with the serialization/deserializtion
 		 * of cglib, sql-timestamp, hibernate collections, etc
 		 */
-        collectionConverter.setConverterLookup(xstream.getConverterLookup());
-        xstream.registerConverter(collectionConverter);
+		if (collectionConverter != null) {
+			collectionConverter.setConverterLookup(xstream.getConverterLookup());
+			xstream.registerConverter(collectionConverter);
+		}
 
-    }
+		try {
+			MethodUtils.invokeExactStaticMethod(SimpleXStreamSerializer.class,
+				"setupXStreamSecurity", xstream, adminService);
```

Comment:
Use reflection as this method will be only available in the latest openmrs-core and we want this module to continue to support older versions (even though they may not be fully secured).